### PR TITLE
feat: [FFM-8870]: Update flag edit dialog to use ModalDialog

### PR DESCRIPTION
**Pull request by knagurski**
This PR updates the Flag Edit dialog to use the `ModalDialog` component.

|Before|After|
|------|------|
|Before|After|

### CODE REVIEW

File: src/modules/75-cf/components/FlagActivation/hooks/useEditFlagDetailsModal.tsx [hunk 1]
```diff
@@ -123,33 +114,49 @@ const useEditFlagDetailsModal = (props: UseEditFlagDetailsModalProps): UseEditFl
...
-                <Text>{getString('cf.editDetails.editDetailsHeading')}</Text>
-
+              <Layout.Vertical spacing="medium">
                 <FormInput.Text name="name" label={getString('name')} />
 
-                <FormInput.TextArea name="description" label={getString('description')} />
+                <FormInput.TextArea
+                  name="description"
+                  label={getString('description')}
+                  textArea={{ style: { minHeight: '5rem' } }}
```
Comment (knagurski) [inline src/modules/75-cf/components/FlagActivation/hooks/useEditFlagDetailsModal.tsx:156]: Annoying that I had to do this, but another module has some global CSS that sets the min-height of all textareas as 200px.

File: src/modules/75-cf/components/FlagActivation/hooks/useEditFlagDetailsModal.tsx [hunk 2]
```diff
@@ -123,33 +114,49 @@ const useEditFlagDetailsModal = (props: UseEditFlagDetailsModalProps): UseEditFl
...
+                <Button
+                  intent="primary"
+                  text={getString('save')}
+                  onClick={submitForm}
+                  variation={ButtonVariation.PRIMARY}
+                  disabled={!isValid}
+                />
+                <Button
+                  text={getString('cancel')}
+                  variation={ButtonVariation.SECONDARY}
```
[the remark this answers is not in the document]
Comment (knagurski) [reply src/modules/75-cf/components/FlagActivation/hooks/useEditFlagDetailsModal.tsx:143]: Well spotted. Didn't realise it was missed in the original button sweep!

File: src/modules/75-cf/components/FlagActivation/hooks/useEditFlagDetailsModal.tsx [hunk 3]
```diff
@@ -123,33 +114,49 @@ const useEditFlagDetailsModal = (props: UseEditFlagDetailsModalProps): UseEditFl
...
+                  onClick={hideEditDetailsModal}
+                />
+              </Layout.Horizontal>
+            }
+          >
             <Form data-testid="edit-flag-form">
```
Comment (crystal-fong) [inline src/modules/75-cf/components/FlagActivation/hooks/useEditFlagDetailsModal.tsx:149]: wonder why is this added when there's no test suite for this file lol

Comment (knagurski) [reply src/modules/75-cf/components/FlagActivation/hooks/useEditFlagDetailsModal.tsx:149]: It's used all over the `FlagDetailsOptionsMenuButton` component's tests

File: src/modules/75-cf/components/FlagActivation/__tests__/__snapshots__/FlagActivationDetails.test.tsx.snap [hunk 4]
```diff
@@ -3,228 +3,151 @@
...
   data-testid="edit-flag-form"
 >
-  <div>
+  <div
+    class="Layout--vertical Layout--layout-spacing-medium StyledProps--main"
+  >
     <div
-      class="Layout--vertical Layout--layout-spacing-large StyledProps--main editDetailsModalContainer"
+      class="bp3-form-group"
     >
```
Comment (RickCraig) [inline src/modules/75-cf/components/FlagActivation/__tests__/__snapshots__/FlagActivationDetails.test.tsx.snap:13]: Just a question... why wouldn't this be:
```html
<div class="bp3-form-group">
```
Is it a style choice that Harness has made? I see a lot of other places there are short elements that take up 3 lines.

**Review by bgribben-harness — APPROVED** [review]

**Review by RickCraig — APPROVED** [review]

File: src/modules/75-cf/components/FlagActivation/__tests__/__snapshots__/FlagActivationDetails.test.tsx.snap [hunk 5]
```diff
@@ -3,228 +3,151 @@
...
+        class="bp3-label"
+        for="name"
       >
-        cf.editDetails.editDetailsHeading
-      </p>
+        name
+         
+        <span
+          class="bp3-text-muted"
+        />
```
Comment (RickCraig) [inline src/modules/75-cf/components/FlagActivation/__tests__/__snapshots__/FlagActivationDetails.test.tsx.snap:22]: Question: Why wouldn't this be:
```html
<span class="bp3-text-muted" />
```
Is it a common Harness thing?

_(edit, moved this onto a line that it was easier to highlight the code)_
